components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java

```diff
@@ -728,19 +729,33 @@ public User updateUserPassword(String code, String password, Property[] properti
         }
         if (notificationsInternallyManaged &&
```

Comment:
There is another method `updateUserPassword` which pass code and  confirmation code.
Need to check how it is used and apply the same fix as below.